File: JavaScript/94.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
    let arr = [];
    let res = [];
    if (root == null) {
        return res;
    }
    let tree = root;
    while (tree != null || arr.length > 0) {
        while (tree != null) {
            arr.push(tree)
            tree = tree.left;
        }
        tree = arr.pop();
        res.push(tree.val)
        tree = tree.right
    }
    return res;
};

File: Java/94.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        LinkedList<Integer> res = new LinkedList<>();
        LinkedList<TreeNode> stack = new LinkedList<>();
        
        if (root == null) {
            return res;
        }
        
        TreeNode tree = root;
        while (tree != null || !stack.isEmpty()) {
            while (tree != null) {
                stack.add(tree);
                tree = tree.left;
            }
            tree = stack.pollLast();
            res.add(tree.val);
            tree = tree.right;
        }
        
        return res;
    }
}